File: Code/07_Pipes_Sinais_Alarmes/Ex7.c
#include <signal.h>
#include <unistd.h>
#include <stdio.h>
#include <stdlib.h>

void tratamento_alarme(int sig)
{
	alarm(1);
	// system("date");
	system("date +%s%3N");
}

int main()
{
	signal(SIGALRM, tratamento_alarme);
	alarm(1);
	printf("Aperte CTRL+C para acabar:\n");
	while(1);
	return 0;
}

File: 2_Programação_sistemas_operacionais/2.4_Pipes_Sinais_Alarmes/Python/Ex7.py
import os, signal

def tratamento_alarme(signum, frame):
	signal.alarm(1)
	# os.system("date")
	os.system("date +%H:%M:%S.%N")

signal.signal(signal.SIGALRM, tratamento_alarme)
signal.alarm(1)
print("Aperte CTRL+C para acabar:")
while True:
	pass